src/System.Private.CoreLib/shared/System/Security/SecureString.cs

```diff
@@ -178,19 +358,122 @@ internal unsafe IntPtr MarshalToString(bool globalAlloc, bool unicode)
             lock (_methodLock)
             {
                 EnsureNotDisposed();
-                return MarshalToStringCore(globalAlloc, unicode);
+
+                UnprotectMemory();
+
+                SafeBuffer? bufferToRelease = null;
+                IntPtr ptr = IntPtr.Zero;
+                int byteLength = 0;
+                try
+                {
+                    Span<char> span = AcquireSpan(ref bufferToRelease).Slice(0, _decryptedLength);
+
+                    if (unicode)
+                    {
+                        byteLength = (span.Length + 1) * sizeof(char);
```

Comment:
I assume this cannot overflow.

Reply:
Right. This cannot overflow because of SecureString cannot be longer than `MaxLength = 65536`.

(And even if it did overflow, the bounds checks would kick in later and avoided any real damage.)